lib/buffer.js

```diff
@@ -20,6 +20,8 @@ function createPool() {
   poolOffset = 0;
 }
 
+// v8::String::kMaxLength defined in v8.h
+const kStringMaxLength = (1 << 28) - 16;
```

Comment:
@bnoordhuis How would you suggest getting the value from C++? If not taking the `require('buffer').kMaxLength` route, then would probably want to use something like `binding.setupBufferJS()`. Though if placed in `internal/buffer` then would have to create a separate C++ call to pull that value out.

I'm not opposed to the change. Just slightly more complex than at face value.